pydra/utils/typing.py

```diff
@@ -636,27 +680,29 @@ def is_subclass(
                 ):
                     return True
             else:
-                if klass is ty.Any:
-                    if ty.Any in candidates:  # type: ignore
-                        return True
-                    else:
-                        return any_ok
-                origin = get_origin(klass)
                 if origin is ty.Union:
-                    args = get_args(klass)
-                    if get_origin(candidate) is ty.Union:
-                        candidate_args = get_args(candidate)
-                    else:
-                        candidate_args = [candidate]
-                    return all(
-                        any(cls.is_subclass(a, c) for a in args) for c in candidate_args
+                    union_args = (
+                        candidate_args if candidate_origin is ty.Union else (candidate,)
                     )
-                if origin is not None:
-                    klass = origin
-                if klass is candidate or candidate is ty.Any:
-                    return True
-                if issubclass(klass, candidate):
-                    return True
+                    matches = all(
+                        any(cls.is_subclass(a, c) for c in union_args) for a in args
+                    )
+                    if matches:
+                        return True
+                else:
+                    if candidate_args and candidate_origin is not ty.Union:
+                        if (
+                            origin
+                            and issubclass(origin, candidate_origin)  # type: ignore[arg-type]
+                            and len(args) == len(candidate_args)
+                            and all(
+                                issubclass(a, c) for a, c in zip(args, candidate_args)
+                            )
+                        ):
+                            return True
```

Comment:
There are a number of these branches that don't seem to be covered by tests. Feels like an invitation to subtle bugs.

Reply:
Nevermind. Not sure why the annotations don't match the codecov report.